### PR TITLE
feat: implemented password successful modal

### LINKED ISSUE #172: [Feature]: SETTINGS PAGE - Implement the Password successful modal
### Description

Implement a modal that appears when a user's password update is successful. 
The modal should include a "Continue" button that, when clicked, disables the modal. 
Additionally, users should be able to click outside the modal to disable it.

### Acceptance Criteria

1. The modal should display a message indicating the password update was successful.
2. The modal should include a "Continue" button that disables the modal when clicked.
3. Clicking outside the modal should also disable it.
4. The modal should be styled according to the provided design specifications.
5. The modal should be responsive across all devices (mobile, tablet, desktop).
6. Unit tests should cover the modal's functionality.
7. Integration tests should ensure the modal works correctly within the settings page.

Success Message Box

### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=10-17157&m=dev

### PR DESCRIPTION
**Pull request by Maiqel1**
# Description

**Closes #172 **

# Changes proposed

## What were you told to do?

Implement the Password successful modal 

## What did you do?

added a PasswordSuccessfulModal.tsx file as well as a PasswordSuccessfulModal.test.tsx file, i used shadcn component for the button and tailwindcss for styling, i also created a mock settings page for testing purposes as the actual page hasnt been implemented yet
Create Next App - Google Chrome 7_18_2024 10_20_12 PM
Create Next App - Google Chrome 7_18_2024 10_20_12 PM

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

**Comment by Maiqel1** [comment]
@ickynavigator , kindly review and merge

File: src/components/modal/PasswordSuccessfulModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { expect, test } from 'vitest';
```
Comment (ickynavigator) [inline src/components/modal/PasswordSuccessfulModal.test.tsx]: vitest globals are already setup so you can take this out

File: src/components/modal/PasswordSuccessfulModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  test('close the modal when continue is clicked', async () => {
+    let modalClosed = false;
+    const handleClose = () => {
+      modalClosed = true;
+    };
+    const { getByText } = render(
+      <PasswordSuccessfulModal show={true} onClose={handleClose} />
+    );
+  
+    await fireEvent.click(getByText('Continue'));
```
Comment (ickynavigator) [inline src/components/modal/PasswordSuccessfulModal.test.tsx]: Consider using the [userevent api](https://testing-library.com/docs/user-event/intro/) instead

File: src/components/modal/page.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+//this is a mock settings page for testing purposes, the actual page hasnt been implemented yet
```
Comment (ickynavigator) [inline src/components/modal/page.tsx]: you can take out the mocks. connect this once your page has been implemented

**Review by ickynavigator — CHANGES_REQUESTED** [review]

**Comment by Maiqel1** [comment]
@ickynavigator 🙏

**Comment by Maiqel1** [comment]
@ickynavigator kindly review again, please 🙏

**Pull request closed without merging**